pkg/protocol/engine/booker/inmemorybooker/booker.go

```diff
@@ -132,15 +115,38 @@ func (b *Booker) Queue(block *blocks.Block) error {
 
 func (b *Booker) Shutdown() {
 	b.TriggerStopped()
-	b.workers.Shutdown()
 }
 
-func (b *Booker) setRetainBlockFailureFunc(retainBlockFailure func(iotago.BlockID, apimodels.BlockFailureReason)) {
-	b.retainBlockFailure = retainBlockFailure
+func (b *Booker) setupBlock(block *blocks.Block) {
+	var unbookedParentsCount atomic.Int32
+	unbookedParentsCount.Store(int32(len(block.Parents())))
```

Comment:
Wouldn't it be better to use reactive.Counter here instead of having a separate variable?

Reply:
Good idea, I didn't think about it! I've just tried to code a version with it and in my opinion is not very readable. I can show you and decide what version is best.